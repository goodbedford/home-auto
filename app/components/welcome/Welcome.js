import React from "react";
import {Route, Link} from "react-router-dom";
import Button from "../button/Button";
const PropTypes = React.PropTypes;

const propTypes = {
  msg: PropTypes.string.isRequired
};
const defaultProps = {
  msg: "welcome"
};
const Welcome = (props) => {

  return (
    <div className="welcome">
      <h1 className="txt txt--header">Home Automation</h1>
      <h3 className="txt txt--sub-header">{props.msg}</h3>
      <div>
        <Link to="/homes">
          <Button txt="homes" />
        </Link>
      </div>
    </div>
  );
};
Welcome.propTypes = propTypes;
Welcome.defaultProps = defaultProps;

export default Welcome;
