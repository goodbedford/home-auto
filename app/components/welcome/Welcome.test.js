/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Welcome from "./Welcome";

describe("<Welcome />", () => {
  const props =  {msg: "welcome"};
  it("Welcome snapshot", () => {
    const wrapper = shallow(<Welcome {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
