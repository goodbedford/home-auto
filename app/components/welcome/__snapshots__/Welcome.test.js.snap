// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> Welcome snapshot 1`] = `
<div
  className="welcome"
>
  <h1
    className="txt txt--header"
  >
    Home Automation
  </h1>
  <h3
    className="txt txt--sub-header"
  >
    welcome
  </h3>
  <div>
    <Link
      replace={false}
      to="/homes"
    >
      <Button
        txt="homes"
      />
    </Link>
  </div>
</div>
`;
